CORE/world_gadgets/data_engine.rb

```diff
@@ -0,0 +1,47 @@
+
+
+class DataEngine
+
+  def initialize(logger=nil)
+    @logger = logger
+  end
+
+  def load_data_from_yml(filename)
+    @logger.debug "Loading data from file #{filename}" if @logger
+    begin
+      data = YAML.load_file(filename)
+      return data
+    rescue Psych::SyntaxError => e
+      raise "ERROR: Tally could not parse this YML file!\n"\
+		        "\tMY GUESS: You have a syntax error in your .yml file\n"\
+		        "\tYML_FILE: #{filename}\n"\
+		        "\tNOTE: Could be from inherited page YML file\n"
+      "\tORIGINAL MESSAGE: #{e.message}\n\n"
+    rescue Errno::ENOENT => e
+      raise "ERROR: Tally could not parse this YML file!\n"\
+                "\tMY FIRST GUESS: The path to this YML file is incorrect.\n"\
+                "\tMY SECOND GUESS: The yml file is missing.\n"\
+                "\tYML_FILE: #{filename}\n"\
+                "\tORIGINAL MESSAGE: #{e.message}\n\n"
+    end
+  end
+
+  def get_yml_data(data_type, filename, data_name)
+    data_name = data_name.upcase.gsub(" ","_")
+    raw_data = load_data_from_yml(filename)[data_type]
+    data = raw_data[data_name]
+    default_data = raw_data["DEFAULT"]
```

Comment:
We could perhaps change lines 35-36 to be:
`return data ? default_data.merge!(data) : default_data`
But It can't really be condensed further than this without altering functionality. (it's wordy because this needs to merge default and named data)